src/nvim/api/autocmd.c

```diff
@@ -72,8 +72,9 @@ static int64_t next_autocmd_id = 1;
 ///             - event (string|array): event or events to match against |autocmd-events|.
 ///             - pattern (string|array): pattern or patterns to match against |autocmd-pattern|.
 ///             Cannot be used with {buffer}
-///             - buffer: Buffer number or list of buffer numbers for buffer local autocommands
+///             - buffer number: Buffer number or list of buffer numbers for buffer local autocommands
 ///             |autocmd-buflocal|. Cannot be used with {pattern}
+///             - id number: Autocommand ID to match.
```

Comment:
while we're here, sort these key names? they are in random order.

Reply:
Having the type immediately after the field name without any separation is strange. It's better to put the type in parentheses. For example, the docstring from `nvim_get_hl` `opts` looks like this:
```c
/// @param opts  Options dict:
///                 - name: (string) Get a highlight definition by name.
///                 - id: (integer) Get a highlight definition by id.
///                 - link: (boolean, default true) Show linked group name instead of effective definition |:hi-link|.
///                 - create: (boolean, default true) When highlight group doesn't exist create it.
```

Reply:
make sense.

Reply:
also update the type  use integer and table instead of number array.